torchmetrics/functional/text/eed.py

```diff
@@ -0,0 +1,436 @@
+
+
+def _preprocess_ja(sentence: str) -> str:
+    """Copied from https://github.com/rwth-i6/ExtendedEditDistance/blob/master/util.py.
+
+    Raises:
+        ValueError: If input sentence is not of a type `str`.
+    """
+    if not isinstance(sentence, str):
+        raise ValueError(f"Only strings allowed during preprocessing step, found {type(sentence)} instead")
+
+    sentence = sentence.rstrip()  # trailing space, tab, newline
+    # characters which look identical actually are identical
+    sentence = unicodedata.normalize("NFKC", sentence)
+    return sentence
+
+
+def _eed_compute(sentence_level_scores: List[Tensor]) -> Tensor:
+    """Final step in extended edit distance.
+
+    Args:
+        sentence_level_scores:
+            list of sentence-level scores as floats
+
+    Return:
+        average of scores as a tensor
+    """
+    if len(sentence_level_scores) == 0:
+        return tensor(0.0)
+
+    average = sum(sentence_level_scores) / tensor(len(sentence_level_scores))
+    return average
+
+
+def _preprocess_sentences(
+    preds: Union[str, Sequence[str]],
+    target: Sequence[Union[str, Sequence[str]]],
+    language: Union[Literal["en"], Literal["ja"]],
+) -> Tuple[Union[str, Sequence[str]], Sequence[Union[str, Sequence[str]]]]:
+    """Proprocess strings according to language requirements.
+
+    Args:
+        preds: An iterable of hypothesis corpus.
+        target: An iterable of iterables of reference corpus.
+        language: Language used in sentences. Only supports English (en) and Japanese (ja) for now. Defaults to en
+
+    Return:
+        Tuple of lists that contain the cleaned strings for target and preds
+
+    Raises:
+        ValueError: If a different language than 'en" or 'ja' is used
+        ValueError: If length of target not equal to length of preds
+        ValueError: If objects in reference and hypothesis corpus are not strings
+    """
+    # sanity checks
+    target, preds = _validate_inputs(hypothesis_corpus=preds, reference_corpus=target)
+
+    # preprocess string
+    if language == "en":
+        preprocess_function = _preprocess_en
+    elif language == "ja":
+        preprocess_function = _preprocess_ja
+    else:
+        raise ValueError(f"Expected argument `language` to either be `en` or `ja` but got {language}")
+
+    preds = [preprocess_function(hyp) for hyp in preds]
+    target = [[preprocess_function(ref) for ref in reference] for reference in target]
+
+    return preds, target
+
+
+def _compute_sentence_statistics(
+    preds_word: str,
+    target_words: Union[str, Sequence[str]],
+    alpha: float = 2.0,
+    rho: float = 0.3,
+    deletion: float = 0.2,
+    insertion: float = 1.0,
+) -> Tensor:
+    """Compute scores for ExtendedEditDistance.
+
+    Args:
+        target_words:
+            An iterable of reference words
+        preds_word:
+            A hypothesis word
+        alpha:
+            optimal jump penalty, penalty for jumps between characters
```

Comment:
```suggestion
            An optimal jump penalty, penalty for jumps between characters
```
pls fix this one and others as well :]